app/services/auth.service.js

```diff
@@ -90,19 +90,23 @@ async function localStrategy2Fa (req, done) {
   }
 }
 
-function localDirectStrategy (req, done) {
-  return userService.findByExternalId(req[INVITE_SESSION_COOKIE_NAME].userExternalId)
-    .then((user) => {
-      lodash.set(req, 'gateway_account.currentGatewayAccountId', lodash.get(user, 'serviceRoles[0].service.gatewayAccountIds[0]'))
-      req.session.secondFactor = 'totp'
-      setSessionVersion(req)
-      req[INVITE_SESSION_COOKIE_NAME].destroy()
-      done(null, user)
-    })
-    .catch(() => {
-      req[INVITE_SESSION_COOKIE_NAME].destroy()
-      done(null, false)
-    })
+async function localDirectStrategy (req, done) {
+  const registrationSession = req[INVITE_SESSION_COOKIE_NAME]
+  try {
+    if (!(registrationSession && registrationSession.userExternalId)) {
+      return done(null, false)
+    }
+
+    const user = await userService.findByExternalId(registrationSession.userExternalId)
+    lodash.set(req, 'gateway_account.currentGatewayAccountId', lodash.get(user, 'serviceRoles[0].service.gatewayAccountIds[0]'))
```

Comment:
Do we need to do this anymore now that users are landing on the services page? I'm guessing this would then be set by the user following a link into an individual account. I might be missing something!

Reply:
Yes I don't think we need this anymore! I think this was possibly left around from when we got rid of the gateway account cookie as a temporary thing but never removed? I can look into this quickly and then maybe raise another PR